src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs

```diff
@@ -16,10 +18,14 @@ public class ConsoleLifetime : IHostLifetime, IDisposable
         private readonly ManualResetEvent _shutdownBlock = new ManualResetEvent(false);
 
         public ConsoleLifetime(IOptions<ConsoleLifetimeOptions> options, IHostingEnvironment environment, IApplicationLifetime applicationLifetime)
+            : this(options, environment, applicationLifetime, NullLoggerFactory.Instance) { }
+
+        public ConsoleLifetime(IOptions<ConsoleLifetimeOptions> options, IHostingEnvironment environment, IApplicationLifetime applicationLifetime, ILoggerFactory loggerFactory)
         {
             Options = options?.Value ?? throw new ArgumentNullException(nameof(options));
             Environment = environment ?? throw new ArgumentNullException(nameof(environment));
             ApplicationLifetime = applicationLifetime ?? throw new ArgumentNullException(nameof(applicationLifetime));
+            Logger = loggerFactory.CreateLogger("Microsoft.Extensions.Hosting.Host");
```

Comment:
`CreateLogger<Host>`?

Reply:
Microsoft.Extensions.Hosting.Host doesn't actually exist, it's Microsoft.Extensions.Hosting.Internal.Host. I didn't want Internal in the log scope.

Reply:
What about https://github.com/aspnet/Extensions/blob/master/src/Hosting/Hosting/src/Internal/Host.cs#L23 then?

Reply:
Yeah, we should re-think that. We've changed that kind of thing elsewhere. @glennc

Reply:
Offline note: making the pubternal types properly internal and flattening the namespace would indirectly deal with this. We've taken that approach in other projects like MVC.